components/server/src/jobs/runner.ts

```diff
@@ -56,6 +56,8 @@ export class JobRunner {
                 frequencyMs: job.frequencyMs,
                 redisLockId: job.lockId,
             });
+            // immediately run the job once
+            this.run(job);
             disposables.push(repeat(() => this.run(job), job.frequencyMs));
```

Comment:
This can cause the first job, and the subsequent periodic to co-run if the first takes long enough. This is fine, as the job itself will acquire the mutex.

Would it make sense to delay the first job by maybe 5 secs to give the server a chance to fully start?

Reply:
> Would it make sense to delay the first job by maybe 5 secs to give the server a chance to fully start?

Would be good to know when there are jobs that have certain dependencies which are not met at server start. I can't think of any atm, so would like to keep it like this to not create any uncertainty around this.

Reply:
Makes sense. I meant more in the sense that at server startup time, a lot of memory and CPU is being used to setup db conns, allocate objects etc. Basically at startup time, we see higher CPU/Mem usage than in the steady state